group_layers/group-layers/src/main/java/com/esri/samples/group_layers/GroupLayersSample.java

```diff
@@ -78,40 +82,59 @@ public void start(Stage stage) {
-      ArcGISSceneLayer devBBuildings = new ArcGISSceneLayer("https://tiles.arcgis.com/tiles/P3ePLMYs2RVChkJx/arcgis/rest/services/DevB_BuildingShells/SceneServer");
-      ArcGISSceneLayer devATrees = new ArcGISSceneLayer("https://tiles.arcgis.com/tiles/P3ePLMYs2RVChkJx/arcgis/rest/services/DevA_Trees/SceneServer");
-      FeatureLayer devAPathways = new FeatureLayer(new ServiceFeatureTable(" https://services.arcgis.com/P3ePLMYs2RVChkJx/arcgis/rest/services/DevA_Pathways/FeatureServer/1"));
-      FeatureLayer devProjectArea = new FeatureLayer(new ServiceFeatureTable("https://services.arcgis.com/P3ePLMYs2RVChkJx/arcgis/rest/services/DevelopmentProjectArea/FeatureServer/0"));
-
-      // create a group layer from scratch by adding the layers as children
-      groupLayer = new GroupLayer();
-      groupLayer.setName("Group: Dev A");
-      groupLayer.getLayers().addAll(Arrays.asList(devATrees, devAPathways, devABuildings));
-
-      // add the group layer and other layers to the scene as operational layers
-      scene.getOperationalLayers().addAll(Arrays.asList(groupLayer, devBBuildings, devProjectArea));
-
-      // zoom to the extent of the group layer when the child layers are loaded
-      groupLayer.getLayers().forEach(childLayer ->
-        childLayer.addDoneLoadingListener(() -> {
-          if (childLayer.getLoadStatus() == LoadStatus.LOADED) {
-            sceneView.setViewpointCamera(new Camera(groupLayer.getFullExtent().getCenter(), 700, 0, 60, 0));
+      // create two group layers, set the visibility modes and add layers
+      projectAreaGroupLayer = new GroupLayer();
+      projectAreaGroupLayer.setName("Project area group");
+      projectAreaGroupLayer.setVisibilityMode(GroupVisibilityMode.INDEPENDENT);
+      projectAreaGroupLayer.getLayers().addAll(Arrays.asList(
+        (new FeatureLayer(new ServiceFeatureTable("https://services.arcgis.com/P3ePLMYs2RVChkJx/arcgis/rest/services/DevelopmentProjectArea/FeatureServer/0"))),
+        (new FeatureLayer(new ServiceFeatureTable("https://services.arcgis.com/P3ePLMYs2RVChkJx/arcgis/rest/services/DevA_Pathways/FeatureServer/1"))),
+        (new ArcGISSceneLayer("https://tiles.arcgis.com/tiles/P3ePLMYs2RVChkJx/arcgis/rest/services/DevA_Trees/SceneServer"))
+      ));
+
+      buildingsGroupLayer = new GroupLayer();
+      buildingsGroupLayer.setName("Buildings group");
+      buildingsGroupLayer.setVisibilityMode(GroupVisibilityMode.EXCLUSIVE);
+      buildingsGroupLayer.getLayers().addAll(Arrays.asList(
+        (new ArcGISSceneLayer("https://tiles.arcgis.com/tiles/P3ePLMYs2RVChkJx/arcgis/rest/services/DevA_BuildingShells/SceneServer")),
+        (new ArcGISSceneLayer("https://tiles.arcgis.com/tiles/P3ePLMYs2RVChkJx/arcgis/rest/services/DevB_BuildingShells/SceneServer"))
+      ));
+
+      // Display an alert if the child layers in the group layers fail to load
+      projectAreaGroupLayer.getLayers().forEach(layer ->
+        layer.addDoneLoadingListener(() -> {
```

Comment:
Perhaps adding a `.loadAsync` above this line (and below when iterating over buildingsGroupLayer) would help with the wait happening when switching between the different group layers

Reply:
@JonLavi The last few days I've tested a load of strategies in terms of how the layers are loaded and also added to the UI (which is currently in the order at which the layers are loaded, rather than the order they are on the map) to see if/what improvements can be made.  @TADraeseke and I ran through it and compared with Android too. Conclusion has been that I've moved the loading out of this method (though the switch speed still seems to just be the layers themselves). I've also left the ordering pattern as it is for the UI.

I've saved the testing on a branch so if you or @Rachael-E  wanted me to share/discuss those just let me know.